documentation/solution-architecture.md

```diff
@@ -0,0 +1,151 @@
+
+The following channels will be created using this solution holding the following information (verifiable claims) with what kind of accessibility retention:
+
+1) public service channel of issuer, destroyed at service shutdown
+  - that a service with some id is attending to needs for what : (ssid, attendTo, what)
+  - (possibly a log of needs that has been attended to)
+2) public customer channel of client, destroyed after solving the need or at max for X minutes or service shutdown whatever comes first
+  - that an anonymous client with some id is in need for what : (ssid, need, what)
+3) private customer channel, destroyed after solving the need or at max for X minutes or service shutdown whatever comes first
+  - that the client refers to a certain BSN number for this. (ssid, BSN, bsn-number)
+  - that the client allows access to this channel to issuer
+  - that the client attests the information prepared by the issuer in the delivery channel to be true
+  - that the client successfully got the signed information in the format it wants by denoting the need as solved
+4) private delivery channel of issuer, destroyed after solving the need or at max for X minutes or service shutdown whatever comes first
+  - that a service with some id is attending to a single specific need that has been noticed for what : (ssid, attendTo, what)
+  - all the information the issuer has prepared to issue, taken from the NLX result channel
+  - that the specific client in need is allowed to access this channel
+  - that it matches a specific need for that same what
+  - that it attests (with signature using NLX outway key) the information to be issued
+5) clerk / auth svc channel, held in memory until read or at max for X minutes or service shutdown whatever comes first
+  - that the client has been identified / authenticated (and authorised) to retrieve the product in relation to the BSN (or other source argument configured) denoted in a given BSN claim in the private customer channel
+6) private NLX channel of issuer, held in memory until read or at max for X minutes or service shutdown whatever comes first
+    - the result of a NLX call stored in the channel of and thus signed with the NLX certificate of the issuer (every claim specifically allowed to specific other actors)
+7) validation channel of validator, destroyed after validation or service shutdown whatever comes first
+      - the same claim and attestation pair as was issued from the delivery channel of the issuer, read from the QR code on a piece of paper
+
```

Comment:
Maybe we should mention the location of the channels? Or where the keys are held?

Reply:
I added a where column which also mentions where private keys reside